x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/utils/NativeMemoryCalculatorTests.java

```diff
@@ -49,6 +53,39 @@ public void testAllowedBytesForMLWhenAutoIsFalse() {
         }
     }
 
+    public void testConsistencyInAutoCalculation() {
+        for (Tuple<Long, Long> nodeAndJvmSize : Arrays.asList(
+            Tuple.tuple(1073741824L, 432013312L), // 1GB and true JVM size
+            Tuple.tuple(2147483648L, 536870912L), // 2GB ...
+            Tuple.tuple(4294967296L, 1073741824L), // 4GB ...
+            Tuple.tuple(8589934592L, 2147483648L), // 8GB ...
+            Tuple.tuple(17179869184L, 2147483648L), // 16GB ...
+            Tuple.tuple(34359738368L, 2147483648L), // 32GB ...
+            Tuple.tuple(68719476736L, 2147483648L) // 64GB ...
```

Comment:
Does this test still pass if you add in some values that are close to the round multiples, for example 0.9GB, 1.1GB, 1.9GB and 2.1GB?  If it doesn't then we need to check with the Cloud team how exact their node sizes are across different Cloud providers.

Also, 58GB is a node size that gets used with one of the IaaS providers I believe.

Reply:
@droberts195 I will add those conditions!

For these, I went through every node size in GCP and looked at the native size and JVM size. I can work on doing that with some other providers